Add more rest config generation options.

briantkennedy opened the pull request:
* Add function that will try to auto connect
* Add function that will connect from inside a pod

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/mdruskin/kubernetes-enterprise-control/41)
<!-- Reviewable:end -->

filmil commented on pkg/client/restconfig/restconfig.go:
```diff
@@ -0,0 +1,55 @@
+// the first successfully created configuration.
+func NewRestConfig() (*rest.Config, error) {
+	switch *flagRestConfigSource {
+	case "minkube":
```
typo: minkube.

briantkennedy replied:
done

filmil commented on pkg/client/restconfig/restconfig.go:
```diff
@@ -0,0 +1,55 @@
+		panic(errors.Errorf("No rest config source named %s", *flagRestConfigSource))
+	}
+
+	for _, configCreator := range []func() (*rest.Config, error){
```
Is this code reachable?  The switch above seems to either return  or panic.  What am I not seeing?

briantkennedy replied:
line 40: case "":

filmil commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 3 of 4 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[pkg/client/restconfig/restconfig.go, line 45 at r1](https://reviewable.io:443/reviews/mdruskin/kubernetes-enterprise-control/41#-Kv3ru818rSNU9GabE93-r1-45:-Kv48AdmiyqGfKtxbffR:be5trej) ([raw file](https://github.com/mdruskin/kubernetes-enterprise-control/blob/f0abc3b986d59be526bbca86936ec47265099cba/pkg/client/restconfig/restconfig.go#L45)):*
<details><summary><i>Previously, briantkennedy (Brian Kennedy) wrote…</i></summary><blockquote>

line 40: case "":
</blockquote></details>

OK

Ah yes..., 
but, care to add `// do nothing` here for those of us who forget how golang handles fallthrough?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mdruskin/kubernetes-enterprise-control/41#-:-Kv48bXNZ7fWBvBdtQmS:bnfp4nl)*
<!-- Sent from Reviewable.io -->

briantkennedy commented:
Review status: 3 of 4 files reviewed at latest revision, 1 unresolved discussion.

---

*[pkg/client/restconfig/restconfig.go, line 45 at r1](https://reviewable.io:443/reviews/mdruskin/kubernetes-enterprise-control/41#-Kv3ru818rSNU9GabE93-r1-45:-Kv4DssFE4ndX7VkYwxi:b-v63iwp) ([raw file](https://github.com/mdruskin/kubernetes-enterprise-control/blob/f0abc3b986d59be526bbca86936ec47265099cba/pkg/client/restconfig/restconfig.go#L45)):*
<details><summary><i>Previously, filmil (Filip Filmar) wrote…</i></summary><blockquote>

OK

Ah yes..., 
but, care to add `// do nothing` here for those of us who forget how golang handles fallthrough?
</blockquote></details>

Changed control flow so this is more obvious and added more logging.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mdruskin/kubernetes-enterprise-control/41)*
<!-- Sent from Reviewable.io -->

filmil commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[pkg/client/restconfig/restconfig.go, line 45 at r1](https://reviewable.io:443/reviews/mdruskin/kubernetes-enterprise-control/41#-Kv3ru818rSNU9GabE93-r1-45:-Kv4EjAsO1mo-jyI38nt:bsqjb9o) ([raw file](https://github.com/mdruskin/kubernetes-enterprise-control/blob/f0abc3b986d59be526bbca86936ec47265099cba/pkg/client/restconfig/restconfig.go#L45)):*
<details><summary><i>Previously, briantkennedy (Brian Kennedy) wrote…</i></summary><blockquote>

Changed control flow so this is more obvious and added more logging.
</blockquote></details>

LGTM

Neat!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mdruskin/kubernetes-enterprise-control/41#-:-Kv4EcgUCMbZ-ru0H2id:bnfp4nl)*
<!-- Sent from Reviewable.io -->
